docs/spec/spec.md

```diff
@@ -3274,3 +3345,276 @@ type Product record {
+
+#### 11.1.3 Engaging DataLoaders
+
+To engage a DataLoader with a GraphQL service, follow the steps discussed in the below sections.
+
+##### 11.1.3.1 Import `graphql.dataloader` Submodule
+
+In order to engage the dataloader with a GraphQL service, the `graphql.dataloader` submodule must be imported. This submodule provides the `DataLoader` object, which is used to batch and cache data loading operations.
+
+###### Example: Importing `graphql.dataloader` Submodule
+
+```ballerina
+import graphql.dataloader;
+```
+
+##### 11.1.3.2 Register DataLoaders to Context via ContextInit Function
+
+Users should register the `DataLoader` objects via the `graphql:ContextInit` function. The `DataLoader` objects are meant to be used per request. Therefore, the `graphql:ContextInit` function is the ideal place to register the `DataLoader` objects. By registering the `DataLoader` objects to the `graphql:Context` object, these objects become accessible to all resolver functions of the GraphQL service.
+
+###### Example: Registering DataLoaders to Context via ContextInit Function
+
+```ballerina
+@graphql:ServiceConfig {
+    contextInit: isolated function (http:RequestContext requestContext, http:Request request) returns graphql:Context {
+        graphql:Context context = new;
+        context.registerDataLoader("bookLoader", new dataloader:DefaultDataLoader(batchBooks));
+        return context;
+    }
+}
+service on new graphql:Listener(9090) {
+    // ...
+}
+```
+
+##### 11.1.3.3 Define the Corresponding `prefetch` Method
+
+To engage the DataLoader with a GraphQL field (let's assume the field name is `foo`), define a corresponding `prefetch` method named `preFoo` in the service, where `Foo` represents the Pascal-cased name of the GraphQL field. The `preFoo` method can include some or all of the parameters from the GraphQL field and must include the `graphql:Context` parameter. Adding the parameters of the GraphQL `foo` field to the `preFoo` method is optional. However, if these parameters are added, the GraphQL Engine will make the same parameter values of the GraphQL field available to the `preFoo` method.
+
+The GraphQL Engine guarantees the execution of the `preFoo` method prior to the `foo` method. By default, the GraphQL engine searches for a method named `preFoo` in the service class before executing the `foo` method. If the method name is different, the user can override the prefetch method name using the [`prefetchMethodName`](#722-prefetch-method-name-configuration) configuration of the `@graphql:ResourceConfig` annotation.
+
+The user is responsible for implementing the logic to collect the keys of the data to be loaded into the `DataLoader` in the `preFoo` method. The GraphQL Engine guarantees the execution of the `preFoo` method prior to the `foo` method. Subsequently, the user can implement the logic to retrieve the data from the `DataLoader` within the `foo` method.
```

Comment:
> The GraphQL Engine guarantees the execution of the `preFoo` method prior to the `foo` method.

This part is duplicated in the previous paragraph and here.

Reply:
Removed the latter one